torch/csrc/autograd/profiler.cpp

```diff
@@ -159,6 +169,16 @@ struct ProfilerThreadLocalState
     }
   }
 
+  void setOrAddRemoteProfiledEvents(std::vector<Event>&& remoteProfiledEvents) {
+    // Lock to serialize access from multiple callback threads.
+    std::lock_guard<std::mutex> guard(remoteProfiledEventsMutex_);
```

Comment:
we should then take remoteProfiledEventsMutex_ in consolidate or just use state_mutex_ here

Reply:
we take state_mutex_ anyways in getEventList, so I guess we could just use state_mutex_?

Reply:
Good point - we can remove this mutex and use state_mutex.

Reply:
Good catch, this can be updated to use state_mutex.